diktat-rules/src/main/kotlin/org/cqfn/diktat/ruleset/rules/UselessSupertype.kt

```diff
@@ -0,0 +1,135 @@
+            startNode.treeParent.removeChild(lastNode)
+        }
+    }
+
+    /**
+     * Method finds pair of identifier supertype and method name else return null
+     * example: super<A>.foo() -> return Pair(A, foo)
+     *          super.foo() -> null
+     * @param node - node of type DOT_QUALIFIED_EXPRESSION
+     *
+     * @return pair of identifier
+     */
+    @Suppress("UnsafeCallOnNullableType")
+    private fun findFunWithSuper(node: ASTNode): Pair<ASTNode, ASTNode>? {
+        return Pair(
+                node.findChildByType(SUPER_EXPRESSION)
+                        ?.findChildByType(TYPE_REFERENCE)
+                        ?.findAllNodesWithSpecificType(IDENTIFIER)
+                        ?.first() ?: return null,
+                node.findChildByType(CALL_EXPRESSION)
+                        ?.findAllNodesWithSpecificType(IDENTIFIER)
+                        ?.first() ?: return null)
+    }
+
+    /**
+     * The method looks in the same file for all super interfaces or a class, in each it looks for methods
+     * that can be overridden and creates a map with a key - the name of the method and value - the number of times it meets
+     *
+     * @param superTypeList - list of identifiers super classes
+     * @param methodsName - name of overrides methods
+     *
+     * @return map name of method and the number of times it meets
+     */
+    @Suppress("UnsafeCallOnNullableType")
+    private fun findAllSupers(superTypeList: List<ASTNode>, methodsName: List<String>): MutableMap<String, Int>? {
+        val fileNode = superTypeList.first().parent({ it.elementType == FILE })!!
+        val superNodesIdentifier = superTypeList.map { it.findAllNodesWithSpecificType(IDENTIFIER).first().text }
+        val superNodes = fileNode.findAllNodesWithCondition({ superClass ->
+            superClass.elementType == CLASS &&
+                    superClass.getIdentifierName()!!.text in superNodesIdentifier
+        }).mapNotNull { it.findChildByType(CLASS_BODY) }
+        if (superNodes.size != superTypeList.size)
+            return null
+        val functionNameMap = mutableMapOf<String, Int>()
+        superNodes.forEach { classBody ->
+            val overrideFunctions = classBody.findAllNodesWithSpecificType(FUN)
+                    .filter {
+                        (if (classBody.treeParent.hasChildOfType(CLASS_KEYWORD)) it.findChildByType(MODIFIER_LIST)!!.hasChildOfType(OPEN_KEYWORD) else true) &&
+                                it.getIdentifierName()!!.text in methodsName
+                    }
+            overrideFunctions.forEach {
+                if (functionNameMap.containsKey(it.getIdentifierName()!!.text))
+                    functionNameMap[it.getIdentifierName()!!.text] = functionNameMap[it.getIdentifierName()!!.text]!! + 1
+                else
+                    functionNameMap[it.getIdentifierName()!!.text] = 1
+            }
+        }
+        return functionNameMap
```

Comment:
```suggestion
        return functionNameMap.toMap()
```
I believe this way you can convert mutable map to immutable, assuming you won't need to change it afterwards.